lib/publiccloud/ecr.pm

```diff
@@ -0,0 +1,44 @@
+# SUSE's openQA tests
+#
+# Copyright © 2021 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: Helper class for Amazon Elastic Container Registry (ECR)
+#
+# Maintainer: Ivan Lausuch <ilausuch@suse.de>, qa-c team <qa-c@suse.de>
+# Documentation: https://docs.aws.amazon.com/AmazonECR/latest/userguide/docker-push-ecr-image.html
+
+package publiccloud::ecr;
+use Mojo::Base 'publiccloud::aws';
+use testapi;
+
+sub push_container_img {
+    my ($self, $name) = @_;
+
+    my $aws_account_id = $self->get_user_id();
+    my $region         = get_var('PUBLIC_CLOUD_REGION',                'eu-central-1');
+    my $repository     = get_var("PUBLIC_CLOUD_CONTAINER_IMAGES_REPO", 'suse-qec-testing');
+    my $tag            = (split /-/, get_required_var("NAME"))[0];
+    my $full_name      = "$aws_account_id.dkr.ecr.$region.amazonaws.com/$repository:$tag";
+
+    $self->{repository} = $repository;
+    $self->{tag}        = $tag;
+
+    assert_script_run("aws ecr get-login-password --region $region | docker login --username AWS --password-stdin $aws_account_id.dkr.ecr.$region.amazonaws.com");
+    assert_script_run("docker tag $name $full_name");
+    assert_script_run("docker push $full_name");
+    return;
+}
+
+sub cleanup {
+    my $self = shift;
+
+    assert_script_run("aws ecr batch-delete-image --repository-name " . $self->{repository} . " --image-ids imageTag=" . $self->{tag});
+    return;
+}
```

Comment:
you overwrite this https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/master/lib/publiccloud/provider.pm#L610 cleanup method here.
So you forget to call `revoke_vault()`.

```suggestion
sub cleanup {
    my $self = shift;

    assert_script_run("aws ecr batch-delete-image --repository-name " . $self->{repository} . " --image-ids imageTag=" . $self->{tag});
    $self->SUPER::cleanup();
}
```

Reply:
done